### ISSUE TITLE
DNS and connectivity issues since v1.72 on Android

### ISSUE THREAD

**Issue by lwbt** [tailscale/tailscale#13352]
### What is the issue?

I'm running Tailscale for Android on my smartphone (Pixel 7). In recent versions I have noticed connectivity issues.

When I open the app I can see that my status is `Connected 🛈`. The 🛈 is shown in red. When I tap on the 🛈 sign, I currently see the following:

> # Health warnings
>
> **Network map response timeout** \
>  Network map response timeout Tailscale hasn't received a network map from the coordination server in 2m7s.

Yesterday it was something like this:

> # Health warnings
>
> can't reach the configured DNS servers

See https://github.com/tailscale/tailscale/issues/13235

I checked https://status.tailscale.com/ and it all seems fine.

- I don't see any issues with my mobile service provider since everything works fine while Tailscale is disabled.
- When I use my exit node my connection is slower but it also works compared to immediate error when loading a website or starting a download while Tailscale VPN is active without an exit note.
- I noticed that occasionally a service works, like using some IoT service (may be that service uses an IP when DNS fails... I don't know).
- This is not limited to one device, I have a Pixel 4 running Tailscale and it has the same issue while using a slightly different MVNO

### Steps to reproduce

- Install Tailscale on Android
- Start the app and join a Tail Net
- Observe health warnings in the app
- Check Internet connectivity by opening a website in a browser, observe timeouts
- Stop the app and disconnect VPN
- Check Internet connectivity by opening a website in a browser, observe that it loads properly
- Start Tailscale and connect to an exit node on you network
- Check Internet connectivity by opening a website in a browser, observe that it loads properly

### Are there any recent changes that introduced the issue?

Version upgrades of the app through Playstore, I guess, I have not changed much or anything at all on the phone in the past weeks. In fact on the secondary phone I'm certain I didn't change anything except letting apps update themselves. I don't recall any changes to my Tail Net configuration or account since these new versions have been released.

### OS

Android

### OS version

14

### Tailscale version

 1.72.0-taa448d5a9-g

### Other software

Not really. My home network uses Ubuquiti Unifi and I don't use the Unifi VPN while I'm on Tailscale.

### Bug report

BUG-f1e603720e8c5a9b95b97c0678c923a5ca45944637cd3262f902342426219d8c-20240903174349Z-1a647668bc7205d1

**Comment by agottardo** [comment]
Just to provide some context here: we added health warnings in 1.72 on Android, and that's the reason you're now able to see these warnings. It's very likely this stuff was already broken in the past, and you have been experiencing these temporary network issues in previous versions as well... you were simply not aware of it.

> can't reach the configured DNS servers

Are you attempting to use a DNS server on your tailnet, or something else?

**Comment by lwbt** [comment]
> we added health warnings in 1.72 on Android, and that's the reason you're now able to see these warnings.

Thank you for this information.

> It's very likely this stuff was already broken in the past, and you have been experiencing these temporary network issues in previous versions as well... you were simply not aware of it.

They seem rather permanent to me. Usual Internet access had not problems before. I cannot pin point to a specific date or version, because before everything seemed to work fine just as advertised for Tailscale. Install, join, have fund with your network where ever you are.

> > can't reach the configured DNS servers
> 
> Are you attempting to use a DNS server on your tailnet, or something else?

No, not that I'm aware of. I configured an exit node and that's all I have done for now.

- Where do I need to look for in the web console for potential issues?
- Can I run any commands/tools on the phones to provide more information?

**Comment by agottardo** [comment]
Can you point me to a bug report identifier **and** a date-time the connectivity issue verified? So I can see in the logs what was happening at that time.

**Comment by lwbt** [comment]
> Can you point me to a bug report identifier **and** a date-time the connectivity issue verified? So I can see in the logs what was happening at that time.

> ### Bug report
> 
> BUG-f1e603720e8c5a9b95b97c0678c923a5ca45944637cd3262f902342426219d8c-20240903174349Z-1a647668bc7205d1

Shouldn't that be the bug report identifier? I noticed the connectivity issues first on 2024-08-29 or may be 2024-08-31.

> You can see the logs yourself by plugging your phone into a computer, and using `adb logcat`.

`adb -d logcat -d > log.txt` give me a file of 3.4 MB, starting at `09-04 00:27:48.901`. Is any of that worth posting? I trid filtering through the noise with `grep -i tailscale log.txt | sort -k3 | less`  which gives me about 20 lines without duplicates.

**Comment by nrsmac** [comment]
Same issue here! I don't think I've ever gotten MagicDNS to work on Android (Pixel 6)

**Comment by lwbt** [comment]
> I don't think I've ever gotten MagicDNS to work on Android

I have discovered Tailscale just a few months ago and I think MagicDNS was working just fine.

> MagicDNS automatically registers DNS names for devices in your network.

https://tailscale.com/kb/1081/magicdns

I was able to stream music with Jellyfin from home while I was on LTE using MagicDNS names.

**Comment by nrsmac** [comment]
Between my laptop and my server, I can point to Tailnet hosts using just the Tailscale hostname, but this is not the case from my Android phone.. I can use the MagicDNS FQDN but not the MagicDNS hostname on its own with ping. I am also getting the same warning described in OP.

**Comment by lwbt** [comment]
> ### Tailscale version
> 
> 1.72.0-taa448d5a9-g

Both of my phones are still running this version. Just a few minutes ago both phones were disconnected and logged out from my tailnet. (I run some monitoring, that's how I got notified immediately.)

- Pixel 4 logged back in with no issues except a message that I was logged out. `Connected ⚠` was shown were there had previously been a `🛈` sign. I force closed the app, started it again and warning no messages were displayed, just `Connected`.
- Pixel 7 had a different message that the relay server in Amsterdam could not be reached. I force closed the app, started it again and warning no messages were displayed, just `Connected`. I tried opening websites through LTE connection and everything was fine. (Note: You have to kill / force close the app, otherwise you will still have the issue. You don't have to wipe the cache or data from the app, just a proper start is enough.)

Let's hope it stays like this and we can close this issue. I would close it in a week or a month if there are no objections.

**Comment by lwbt** [comment]
> # Tailscale v1.74.0
> **Android**
> Fixed: [Tailscale DNS](https://tailscale.com/kb/1054/dns) works as expected when switching between Wi-Fi and cellular networks.

https://tailscale.com/changelog#2024-09-12

Hmm okay. My version is still at 1.72.

**Comment by kari-ts** [comment]
Please reopen if you encounter issues again. Thanks!

**Comment by R9anSmart** [comment]
Getting this issue on pixel 7 Pro. Any help available?

**Comment by jradxl** [comment]
Just thought I'd mention that a friend and I are both getting this message since we've both just got new Samsung S24FEs, which are Android 14. My older Smasung S6 tablet using Android 13 doesn't show this error message.

**Comment by nickoneill** [comment]
@jradxl are you seeing an issue with DNS resolution or is this strictly an issue with the warning not going away?

**Comment by R9anSmart** [comment]
Both, sometimes the warning shows and the internet is fine. Other times it needs turning off in order to restore service

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by jongrall** [comment]
This issue is still occurring for me in the official Tailscale client v1.90.9 on Android 16 on a Google Pixel 9a.

Symptoms are the same as what has already been reported:

- Occasional warnings about DNS being unavailable (I configured Global DNS override in the Tailscale Admin console to point to NextDNS).

- Occasional warnings about specific relay servers being unavailable, even ones that are not near me. Upon checking the Tailscale status website at the same time to see if there is an outage, it appears that there is no problem with said relay servers. tailscale netcheck on my linux machines confirms that.

- Occasional network map response timeouts. Oddly enough, the Android Tailscale client always says that Tailscale hasn't received a network map from the coordination server in 2m7s. This amount of time never changes.

- Occasional warnings about being unable to reach the coordination server, and that tailnet connectivity may become degraded. 

Only fix is to disconnect from Tailscale in the Android client, force quit the app, relaunch and reconnect. Then the warnings are gone, only to return a few hours or days later.

Of note: these issues do not occur on any of my other tailnet devices, including those on the same network as my Android device. Also, I have checked that devices are able to directly connect (even to my Android device, occasionally) using tailscale ping and tailscale status (i.e. without needing a relay server). I've also added firewall rules on my network gateway to allow internal devices to start UDP connections from port 41641 to anywhere:anyport to help facilitate direct connections. I've whitelisted connections to tailscale's public coordination and login servers and ts.net. None of that seems to help. These problems also seem to occur when I'm on a cellular network, and not behind any firewall that I control, so it seems that any firewall rules on my end to attempt to mitigate this problem are moot.

@kari-ts please reopen this issue, as it appears that you closed it.

**Comment by MarcProe** [comment]
Exactly the same issues here on Pixel 8a Android 16 TailScale 1.90.9 . The problems occurs on different wireless networks, and on cellular network.

**Comment by jongrall** [comment]
To be clear, these problems have been occurring since well before v1.90.9 of the Tailscale Android client. I just wanted to mention that they are still present in the latest version.

**Comment by lwbt** [comment]
I confirm that this issue has not been solved. For a while, around the time it was closed, it looked like it has gotten better, or I wanted to believe that, but it hasn't changed and connection issues as described appear regularly and have to be troubleshooted as described, which is a sub-par user experience.

My old device was a Pixel 7, my current device is a Pixel 10.

I just checked, I have no permission to re-open this issue. Eventually we should create a new issue. I'd prefer that someone else would opens it.

**Comment by kari-ts** [comment]
Thanks for letting us know. As was done [here](https://github.com/tailscale/tailscale/issues/13352#issuecomment-2331474035), could you please send us a bug report and note the rough times when you are experiencing this issue please?

**Comment by jongrall** [comment]
@kari-ts OK, it happened again. Usually the issue occurs more frequently, but this time almost a week before I noticed it.

Noticed at 11:01pm Pacific Standard Time when I opened the Tailscale app. Internet connectivity was affected despite having full 5G signal - sites were not loading (I have DNS override enabled - goes to NextDNS - and the Tailscale app appears to have been in a bad state).

BUG-17fcf75da3e971adc6a990fb01587058925189eadd803a62cf51bca2beaf9d0b-20260116070004Z-286376c97b92dbfe

<img width="1080" height="2424" alt="Image" src="https://github.com/user-attachments/assets/bfca03bf-db25-4c6e-baa0-97053fe45f00" />

Internet connectivity then began working again (I didn't change anything, just waited a few minutes), but the Tailscale app still showed the same warnings.

By first toggling off Tailscale in the app, then force-quitting the app, relaunching, and tapping "Connect," the warnings then disappear and things appear to be back to normal (for now).

**Comment by f-gillmann** [comment]
Also have the same problem, force closing the App and reconnecting fixes it.

Happened just now as I'm writing this.

Edit: Sometimes it also begins working again after a few minutes but the warnings persistent exactly like in the comment above

```
BUG-a6d2524cbb85d988ea69407568ee58a1f45c33672850dc94ca89dc16da0d79d2-20260122094300Z-ce2035f19e1abd93
```

<img width="1344" height="2992" alt="Image" src="https://github.com/user-attachments/assets/d8d667b4-8c1e-4117-aad8-f1d91ac97e86" />

**Comment by 0ranki** [comment]
I've also been experiencing this issue for quite a while. Currently running Tailscale 1.94.2 on GrapheneOS, Pixel 7a.

Just now I force killed the Tailscale app, and only after minutes the coordination server error reappeared. I've also had warnings about DNS. Managed to catch this in the logs, here's what's happening around the time the error appeared:

```
...
--------- switch to events
02-28 05:15:36.504 14014 14014 I auditd  : avc=type=1400 audit(0.0:81379): avc:  denied  { bind } for  comm="ip" scontext=u:r:untrusted_app:s0:c210,c256,c512,c768 tcontext=u:r:untrusted_app:s0:c210,c256,c512,c768 tclass=netlink_route_socket permissive=0 bug=b/155595000 app=com.tailscale.ipn
--------- switch to main
02-28 05:15:36.504 14014 14014 W ip      : type=1400 audit(0.0:81379): avc:  denied  { bind } for  scontext=u:r:untrusted_app:s0:c210,c256,c512,c768 tcontext=u:r:untrusted_app:s0:c210,c256,c512,c768 tclass=netlink_route_socket permissive=0 bug=b/155595000 app=com.tailscale.ipn
02-28 05:15:36.545 14015 14015 W libc    : Access denied finding property "ro.debuggable"
02-28 05:15:36.561 14015 14015 W libc    : Access denied finding property "ro.debuggable"
02-28 05:15:38.756 13280 13355 D gojni   : 16.9M/341.9M health(warnable=not-in-map-poll): error: Unable to connect to the Tailscale coordination server to synchronize the state of your tailnet. Peer reachability might degrade over time.
02-28 05:15:43.770 13280 13323 D Health  : Health updated: [not-in-map-poll]
02-28 05:15:43.771 13280 13355 D gojni   : Health: Health updated: [not-in-map-poll]
02-28 05:15:43.773 13280 13323 D Health  : Adding health warning: not-in-map-poll
02-28 05:15:43.774 13280 13314 D gojni   : Health: Adding health warning: not-in-map-poll
02-28 05:16:02.543 14018 14018 W libc    : Access denied finding property "ro.debuggable"
02-28 05:16:02.551 14018 14018 W libc    : Access denied finding property "ro.debuggable"
--------- switch to events
...
```
Bug report ID BUG-cb371136cb16f834e54e4ef939810413d23184a2c64bae423ba010115ebf7039-20260228054235Z-6e83123b8ceca8e7


The `netlink_route_socket` SELinux error and `ro.debuggable` errors appear a lot in the logs. I don't have the GrapheneOS Exploit protection compatibility mode enabled for Tailscale.

**Comment by jongrall** [comment]
@0ranki I too am running GrapheneOS, and like you I also don't have exploit protection compatibility mode enabled for Tailscale. In fact shortly after installation time I gave Tailscale unrestricted "App battery usage", and "Manage app if unused" is also off. While I suspect custom ROM users are more likely to use Tailscale and Github than stock Android users, and thus more likely to be on this thread, it's also possible that this issue is specific to GrapheneOS or custom ROMs in general.

The other people on this thread experiencing this bug all reported that they were running Android (as old as version 14, as new as 16) but didn't specify if they were using a plain vanilla release, or a custom ROM. It's likely that Tailscale's own testing of custom ROMs is very limited compared to stock Android.

@lwbt @nrsmac @R9anSmart @MarcProe @f-gillmann could you please chime in with some more details of your Android installs and Tailscale app permissions to help triangulate this issue a little better?

**Comment by lwbt** [comment]
> @lwbt @nrsmac @R9anSmart @MarcProe @f-gillmann could you guys chime in with some more details of your Android installs and Tailscale app permissions to help triangulate this issue a little better?

Stock Android on Pixel 10 and previously Pixel 7. Gave Tailscale the permissions it asked for. Which are apparently only Notifications, according to the app permissions dialog.

**Comment by f-gillmann** [comment]
> [@lwbt](https://github.com/lwbt) [@nrsmac](https://github.com/nrsmac) [@R9anSmart](https://github.com/R9anSmart) [@MarcProe](https://github.com/MarcProe) [@f-gillmann](https://github.com/f-gillmann) could you please chime in with some more details of your Android installs and Tailscale app permissions to help triangulate this issue a little better?

Stock Android 16/17 on Pixel 9. Tailscale has all the permissions and I allowed unrestricted background usage for the app. Using the latest 1.94.2 apk from the GitHub release.

**Comment by hcording** [comment]
> This issue is still occurring for me in the official Tailscale client v1.90.9 on Android 16 on a Google Pixel 9a.
> 
> Symptoms are the same as what has already been reported:
> 
> * Occasional warnings about DNS being unavailable (I configured Global DNS override in the Tailscale Admin console to point to NextDNS).
> * Occasional warnings about specific relay servers being unavailable, even ones that are not near me. Upon checking the Tailscale status website at the same time to see if there is an outage, it appears that there is no problem with said relay servers. tailscale netcheck on my linux machines confirms that.
> * Occasional network map response timeouts. Oddly enough, the Android Tailscale client always says that Tailscale hasn't received a network map from the coordination server in 2m7s. This amount of time never changes.
> * Occasional warnings about being unable to reach the coordination server, and that tailnet connectivity may become degraded.
> 
> Only fix is to disconnect from Tailscale in the Android client, force quit the app, relaunch and reconnect. Then the warnings are gone, only to return a few hours or days later.
> 
> Of note: these issues do not occur on any of my other tailnet devices, including those on the same network as my Android device. Also, I have checked that devices are able to directly connect (even to my Android device, occasionally) using tailscale ping and tailscale status (i.e. without needing a relay server). I've also added firewall rules on my network gateway to allow internal devices to start UDP connections from port 41641 to anywhere:anyport to help facilitate direct connections. I've whitelisted connections to tailscale's public coordination and login servers and ts.net. None of that seems to help. These problems also seem to occur when I'm on a cellular network, and not behind any firewall that I control, so it seems that any firewall rules on my end to attempt to mitigate this problem are moot.
> 
> [@kari-ts](https://github.com/kari-ts) please reopen this issue, as it appears that you closed it.

I am facing this exact issue. The internet connectivity on my device (Pixel 9 Pro, Graphene OS) is fairly inconsistent. Sometimes I'll go a week without issues, sometimes I struggle for a while to get a connection working. The warning are always there though, mostly the coordination server warning.

I use the VPN in always on mode and with the 'block connections without VPN' flag enabled. Could that be a cause?
